src/ConnectCore.js

```diff
@@ -276,6 +278,122 @@ class ConnectCore {
   }
 
   /**
+   *  Wraps up functionality to allow you to easily manage the entire req/res lifecycle
+   *  with a request token created on you server and then fetching the response
+   *  from your server. Given a token this will use the default uriHandler as usual
+   *  or will use a given uriHandler. If given a pollUrl for the request, it will
+   *  poll for a response and return the body. You should form the pollUrl as
+   *  necessary, likely including the pairId (of th challenge) for the request as a param. If a callbackUrl
+   *  is not given, it will pass a uri to the uriHandler, but will not poll for a response.
+   *  If not polling and using the default uriHandler, you will have to close the QR
+   *  code yourself using QRUtil.closeQR().
+   * TODO  update explanation
+   *
+   *  @param    {Object}     txobj             transaction object
+   *  @param    {String}     callbackUrl       application callback url
+   *  @return   {Promise<Object, Error>}       A promise which resolves with a resonse object or rejects with an error.
+   */
+  sendRequestToken({token, pollUrl, postUrl,  uriHandler = this.uriHandler, pollingInterval = 1000}) {
+    const uri = paramsToUri({to: 'me', requestToken: token})
+
+    let isCancelled = false
```

Comment:
@zachferland Instead of duplicating code from `topicFactory` here just add an optional `url` param `newTopic` instead that overrides the url generation.

You can decode the token to pull out the callback_url and the figure out what kind of request it is so you can set the topic type on the call to topic factory as well as handle the response correctly.